libsmtutil/genz3wrapper.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python
+
+import sys
+import re
+
+# Patterns to match Z3 API entry point definitions.
+def_pat = re.compile(" *def_API(.*)")
+extradef_pat = re.compile(" *extra_API(.*)")
+# Pattern to extract name and arguments from the above.
+def_args_pat = re.compile("\('([^']*)'[^\(\)]*\((.*)\)\s*\)")
+# Pattern to extract a list of arguments from the above.
+arg_list_pat = re.compile("[^\(]*\([^\)]*\)[, ]*")
+
+def generateEntryPoint(line, args):
+    m = def_args_pat.match(args)
+    if not m:
+        raise Exception('Could not parse entry point definition: ' + line)
+    name = m.group(1)
+    num_args = len(arg_list_pat.findall(m.group(2)))
+    arglist = ', '.join(f"_{i}" for i in range(num_args))
+    paramlist = ', '.join(f"ArgType<&{name}, {i}> _{i}" for i in range(num_args))
+    print(f'ResultType<&{name}> Z3_API {name}({paramlist})')
+    print('{')
+    print(f'\tstatic auto sym = reinterpret_cast<decltype(&{name})>(Z3Loader::get().loadSymbol(\"{name}\"));')
+    print(f'\treturn sym({arglist});')
+    print('}')
+
+
+print(r"""// This file is auto-generated from genwrapper.py
+#include <z3.h>
+#include <tuple>
+#include <libsmtutil/Z3Loader.h>
```

Comment:
Nitpick, but isn't the ordering we use: local subdirectory, project-level, imported projects, system?